src/frontend/next/src/components/SearchBar.tsx

```diff
@@ -0,0 +1,152 @@
+import { MouseEvent } from 'react';
+import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
+import SearchIcon from '@material-ui/icons/Search';
+import {
+  Grid,
+  MenuItem,
+  TextField,
+  FormControl,
+  Paper,
+  IconButton,
+  Box,
+  Typography,
+} from '@material-ui/core';
+
+import SearchInput from './SearchInput/SearchInput';
+import SearchHelp from './SearchHelp';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      overflow: 'visible',
+      maxWidth: '785px',
+      padding: 0,
+      marginTop: '10rem',
+      marginLeft: 'auto',
+      marginRight: 'auto',
+      marginBottom: theme.spacing(6),
+    },
+    card: {
+      padding: theme.spacing(2, 4, 2, 4),
+      backgroundColor: theme.palette.background.default,
+    },
+    header: {
+      padding: 0,
+      marginBottom: theme.spacing(2),
+      backgroundColor: theme.palette.primary.main,
+    },
+    h1: {
+      display: 'block',
+      transition: 'all linear 350ms',
+      fontWeight: 600,
+      color: theme.palette.text.secondary,
+      [theme.breakpoints.between('xs', 'sm')]: {
+        fontSize: '3rem',
+      },
+      [theme.breakpoints.between('md', 'lg')]: {
+        fontSize: '4rem',
+      },
+      [theme.breakpoints.up('xl')]: {
+        fontSize: '5rem',
+      },
+    },
+    iconButton: {
+      backgroundColor: theme.palette.secondary.main,
+      '&:hover': {
+        backgroundColor: theme.palette.secondary.dark,
+      },
+      '& * > .MuiSvgIcon-root': {
+        fontSize: '2rem',
+        color: theme.palette.primary.contrastText,
+      },
+      margin: 0,
+      position: 'absolute',
+      right: '10px',
+      top: '6px',
+    },
+    selectControl: {
+      '& > *': {
+        fontSize: '1.2rem',
+        textTransform: 'capitalize',
+        color: theme.palette.primary.main,
+      },
+    },
+    selectItem: {
+      fontSize: '1.4rem',
+      textTransform: 'capitalize',
+      color: theme.palette.primary.main,
+    },
+  })
+);
+
+type searchBarProps = {
+  text: string;
+  onTextChange: Function;
```

Comment:
for `onTextChange` should we be clearer than just Function. It is an onChange event handler and I believe in TS you have ChangeEventHandler/EventHandler type too. Whats your thought?

Reply:
It should be changed to a ChangeEvent function that returns void
Since it's for an input element, we can type it to `HTMLInputElement`

Reply:
Let's do it in a follow-up PR, because we are drilling a lot of stuff here. After getting everything working we will need to create an interface to `SearchPage`'s components and a `context.` These functions are connected to how we build the URL query of a search. So I believe that we need to be care full with that. It's working now. Let's keep this for now. 

I'm opening an issue for the next step (interface and context) now.

Reply:
https://github.com/Seneca-CDOT/telescope/issues/1584